thrift/cadence.thrift

```diff
@@ -391,6 +405,7 @@ service WorkflowService {
       5: shared.DomainNotActiveError domainNotActiveError,
       6: shared.LimitExceededError limitExceededError,
       7: shared.ClientVersionNotSupportedError clientVersionNotSupportedError,
+      8: shared.WorkflowExecutionAlreadyCompletedError workflowExecutionAlreadyCompletedError,
```

Comment:
Resetting a completed workflow is a valid use case. Is there a scenario where we might want to throw this error from Reset endpoint?